proposals/drafts/elections.rst

```diff
@@ -0,0 +1,669 @@
+
+type
+    **optional**
+    Enumerated among:
+
+    * ballot-measure: A catch-all for generic types of non-candidate-based contests.
+    * initiative: These are usually citizen-driven measures to be placed on the ballot. These could include both statutory changes and constitutional amendments.
+    * referendum: These could include measures to repeal existing acts of legislation, legislative referrals, and legislatively-referred state constitutional amendments.
+    * other: Anything that does not fall into the above categories.
+
+other_type
+    **optional**
+    Allows for cataloging a new type of ballot measure option, when type is specified as "other" (string).
+
+
+CandidateContest
+----------------
+
+A subclass of ``ContestBase`` for repesenting a contest among candidates competing for election to a public office.
+
+id
+    Open Civic Data-style id in the format ``ocd-candidatecontest/{{uuid}}``.
+
+filing_deadline
+    **optional**
+    Specifies the date and time when a candidate must have filed for the contest for the office (datetime).
+
+is_runoff
+    Indicates a contest to decide a prior contest that ended with no candidate receiving a majority of the votes (boolean).
+
+is_unexpired_term
+    Indicates that the former public office holder vacated the post before serving a full term (boolean).
+
+number_elected
+    **optional**
+    Number of candidates that are elected in the contest, i.e. 'N' of N-of-M (integer).
+
+post_ids
+    **repeating**
+    References to the OCD ``Posts`` representing the public offices for which the candidates are competing. If multiple, the primary post should be listed first, e.g., the id for the President post should be listed before the id for Vice-President.
+
+party_id
+    **optional**
+    If the contest is among candidates of the same political party, e.g., a partisan primary election, reference to the OCD ``Party`` representing that political party.
+
+
+PartyContest
+------------
+
+A subclass of ``ContestBase`` which describes a contest in which the possible ballot selections are all political parties. These could include contests in which straight-party selections are allowed, or party-list contests (although these are more common outside of the United States).
+
+id
+    Open Civic Data-style id in the format ``ocd-partycontest/{{uuid}}``.
+
+
+RetentionContest
+----------------
+
+A subclass of ``BallotMeasureContest`` that represents a contest where a candidate is retained in a position, e.g. a judicial retention or recall election.
+
+id
+    Open Civic Data-style id in the format ``ocd-retentioncontest/{{uuid}}``.
+
+candidacy_id
+    Reference to the OCD ``Candidacy`` of the person who will either retain or lose a ``Post`` as a result of the contest.
+
+post_id
+    Reference to the OCD ``Post`` representing the public office the candidate will either retain or lose as a result of the contest.
+
```

Comment:
I think it would be nice to inherit BallotMeasureThreshold for RetentionContest (if it's not already being inherited). Strikes me that some retention contests are likely to have non-50%+1 thresholds.